### PR TITLE
Improve landing page

### PR DESCRIPTION
**Pull request by mrdrogdrog**
### Description
This PR
  - changes the behavior for links so they don't open new tabs
  - improves the title of the documentation header link
  - Rephrases the introduction section
  - Adds two new (for us obvious, but maybe not for others) core features (web-based and self hosted)

### Steps

- [x] Added changes
- [x] I read the [contribution documentation](https://github.com/hedgedoc/hedgedoc.github.io/blob/main/CONTRIBUTING.md) and signed-off my commits to accept the DCO.

### CODE REVIEW

**Review by DerMolly — APPROVED** [review]

**Comment by davidmehren** [comment]
Tbh, I really don't like that there is no "get started" button above the fold anymore.
[All](https://nextcloud.com/) [these](https://rocket.chat/) [other](https://www.discourse.org/) [open](https://bigbluebutton.org/) [source](https://www.drupal.org/) [projects](https://typo3.org/) either have a big button on the main page or in the navbar.

**Comment by ErikMichelson** [comment]
Especially when leaving out the CTA-Buttons, I'm unsure about the label "Download" in the navigation bar.

There is no hint that HedgeDoc is a self-hostable web application within the first screen height. As a user not knowing anything about HedgeDoc, I might try the download page and be irritated what I get there.

**Comment by mrdrogdrog** [comment]
But this problem is also present if you have the CTA buttons. I guess we should rephrase the first section then.

**Comment by mrdrogdrog** [comment]
I just noticed that it's not written anywhere on the landing page.

**Comment by mrdrogdrog** [comment]
Okay. I changed the first two sections a bit. I added these points "HedgeDoc is self hosted" - "HedgeDoc is web based".
However I'm still not very happy with the text for the "self host" feature and actually I don't want to use the firefox icon for "web based". But I don't know what else to use. Any idea?

File: data/en/features.yml [hunk 1]
```diff
@@ -6,6 +6,14 @@
 corefeatures:
   title: "Our core features"
   columns:
+    - title: "Web-based"
+      icon: "fa fa-firefox"
+      content: "Your users don't need extra software. HedgeDoc runs in their browser."
```
Comment (davidmehren) [inline data/en/features.yml:11]:
```suggestion
      content: "You don't need extra software. HedgeDoc runs in the browser."
```

File: data/en/features.yml [hunk 2]
```diff
@@ -6,6 +6,14 @@
...
+
+    - title: "Self hosted"
```
Comment (davidmehren) [inline data/en/features.yml:13]:
```suggestion
    - title: "Self-hosted"
```

File: data/en/features.yml [hunk 3]
```diff
@@ -6,6 +6,14 @@
...
+      icon: "fa fa-server"
+      content: "Keep the data where they belong. On your server."
```
Comment (davidmehren) [inline data/en/features.yml:15]:
```suggestion
      content: "Stay in control of your data. Keep it on your server."
```

**Comment by mrdrogdrog** [comment]
About the icon: how about https://forkaweso.me/Fork-Awesome/icon/html5/ ?

**Review by davidmehren — CHANGES_REQUESTED** [review]

File: data/en/features.yml (hunk 1 above)
Comment (mrdrogdrog) [reply data/en/features.yml:11]: I also thought about this but that sounds a bit like "You don't need to install anything. Not even the software itself! Just use your browser and it magically works!". But on the other hand the next point is "HedgeDoc runs on your server." Sooo I'm not sure.

Comment (mrdrogdrog) [reply data/en/features.yml]: (I changed it for now)

Comment (mrdrogdrog) [reply data/en/features.yml]: How about "The editor runs in the browser" "The markdown editor runs in the browser" ?

Comment (mrdrogdrog) [reply data/en/features.yml]: Changed it to "The editor runs in the browser"

**Review by DerMolly — APPROVED** [review]

**Review by davidmehren — APPROVED** [review]

File: data/en/features.yml [hunk 4]
```diff
@@ -6,6 +6,14 @@
 corefeatures:
   title: "Our core features"
   columns:
+    - title: "Web-based"
+      icon: "fa fa-html5"
+      content: "You don't need extra software. The editor runs in the browser."
```
Comment (ErikMichelson) [inline data/en/features.yml:11]: Maybe "runs in your browser"?

(Avoiding repeating _the_)

Comment (mrdrogdrog) [reply data/en/features.yml:11]: Makes sense

File: content/english/_index.html [hunk 5]
```diff
@@ -4,31 +4,30 @@
...
-With HedgeDoc you can easily collaborate on notes, graphs and even presentations in real-time.
-All you need to do is to share your note-link to your co-workers, and they're ready to go.
+HedgeDoc (formerly known as CodiMD) is an open-source, web-based, self-hosted, collaborative markdown editor.
 
-{{< button external=true label="Getting started" link="https://docs.hedgedoc.org/setup/getting-started" >}}
-{{< button external=true label="Visit the demo" link="https://demo.hedgedoc.org/" outline=true >}}
+You can use it to easily collaborate on notes, graphs and even presentations in real-time.
+All you need to do is to share your note-link to your co-workers and they're ready to go.
+
+{{< button external=true label="Read the install guide" link="https://docs.hedgedoc.org/setup/getting-started" >}}{{< button label="Try the demo" link="https://demo.hedgedoc.org/" outline=true >}}
```
[the remark this answers is not in the document]
Comment (mrdrogdrog) [reply content/english/_index.html]: fixed

**Review by ErikMichelson — APPROVED** [review]

**Pull request merged**